packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx

```diff
@@ -18,7 +18,18 @@ function DefaultChartsAxisTooltipContent(props: ChartsAxisContentProps) {
   if (dataIndex == null) {
     return null;
   }
-  const axisFormatter = axis.valueFormatter ?? ((v) => v.toLocaleString());
+
+  const utcFormatter = (v: string | number | Date): string => {
```

Comment:
The value can also be `null`. Maybe we can move this function definition outside of the component since it does not require any prop

Reply:
if you check the previous implementation null case of v was not considered

https://github.com/mui/mui-x/blob/af046548d4534dcf690efe423e1be0d2acdb91fd/packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx#L21

Reply:
Effectively, I forgot this `axisValue != null && !axis.hideTooltip` condition